Modified custom_arg message_id for compatibility with SendGrid webhook

joshkersey opened the pull request:
Based our discussion in #108.

According to SendGrid support, the `Message-ID` and `smtp-id` are both used by SendGrid to set their unique message ID. Based on the conversation with SendGrid support we should use a custom arg with a key named something other than `Message-ID` or `smtp-id`. 

Tried to set this up to ensure backwards compatibility for users who upgrade so that the `smtp-id` still gets referenced if the `anymail-id` is not present in the webhook data.

medmunds commented on anymail/backends/sendgrid.py:
```diff
@@ -114,10 +114,8 @@ def ensure_message_id(self):
-        # record, we can supply the 'smtp-id' field for any events missing it.
-        self.data.setdefault("custom_args", {})["smtp-id"] = self.message_id
+        # Set anymail-id to track our custom message ID
+        self.data.setdefault("custom_args", {})["anymail-id"] = self.message_id
```
I'd maybe go with "anymail_id" (underscore), to parallel all other webhook data fields. ("smtp-id" is the outlier.)

medmunds commented on anymail/webhooks/sendgrid.py:
```diff
@@ -69,10 +69,12 @@ def esp_to_anymail_event(self, esp_event):
             metadata = {}
 
+        anymail_id = esp_event.get('anymail-id', None)
+        smtp_id = esp_event.get('smtp-id', None)
```
This could be simplified to `message_id = esp_event.get('anymail_id', esp_event.get('smtp-id'))`.
(Use 'anymail_id' if present, else use 'smtp-id' for backwards compatibility.)

joshkersey commented:
Updated based on further discussion in #108. Removed a concern for `Message-ID` from the SendGrid and SendGrid V2 backends in favor of setting a UUID as `anymail_id` for tracking webhook notifications.
